### PR TITLE
tmux-sessionizer: init at 0.2.0

### PR DESCRIPTION
**Pull request by vinnymeller**
###### Description of changes
Adds tmux-sessionizer package and vinnymeller as the maintainer

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/tmux-sessionizer/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "tmux-sessionizer";
+  version = "v0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "jrmoulton";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-FSh3ww5KpktNSvqB3kcIaTxNHypwsVTHze0mgBtuJQE=";
+    };
```
Comment (devusb) [inline pkgs/tools/misc/tmux-sessionizer/default.nix:17]:
```suggestion
  };
```

File: pkgs/tools/misc/tmux-sessionizer/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  cargoSha256 = "sha256-VMdXlBRzM+p7EsU3Wh9Lpg4V1KACX4DUNTh4ARlDwOc=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
```
Comment (devusb) [inline pkgs/tools/misc/tmux-sessionizer/default.nix:22]: to fix build on Darwin:
```suggestion
  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ Security ];
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -12808,6 +12808,8 @@ with pkgs;
 
   tmux-mem-cpu-load = callPackage ../tools/misc/tmux-mem-cpu-load { };
 
+  tmux-sessionizer = callPackage ../tools/misc/tmux-sessionizer { };
```
Comment (devusb) [inline pkgs/top-level/all-packages.nix:12811]:
```suggestion
  tmux-sessionizer = callPackage ../tools/misc/tmux-sessionizer {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

File: pkgs/tools/misc/tmux-sessionizer/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "The fastest way to manage projects as tmux sessions";
+    license = licenses.mit;
+    maintainers = with maintainers; [ vinnymeller ];
+    };
```
Comment (devusb) [inline pkgs/tools/misc/tmux-sessionizer/default.nix:28]:
```suggestion
  };
```

File: pkgs/tools/misc/tmux-sessionizer/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, openssl
+, pkg-config
+}:
```
Comment (devusb) [inline pkgs/tools/misc/tmux-sessionizer/default.nix:6]:
```suggestion
{ lib
, fetchFromGitHub
, stdenv
, rustPlatform
, openssl
, pkg-config
, Security
}:
```

File: pkgs/tools/misc/tmux-sessionizer/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "tmux-sessionizer";
+  version = "v0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "jrmoulton";
+    repo = pname;
+    rev = version;
```
Comment (devusb) [inline pkgs/tools/misc/tmux-sessionizer/default.nix:17]:
```suggestion
  version = "0.2.0";

  src = fetchFromGitHub {
    owner = "jrmoulton";
    repo = pname;
    rev = "v${version}";
```

**Comment by devusb** [comment]
Result of `nixpkgs-review pr 221798` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tmux-sessionizer</li>
  </ul>
</details>

**Review by devusb — APPROVED** [review]

**Comment by vinnymeller** [comment]
@devusb ty!

File: pkgs/tools/misc/tmux-sessionizer/default.nix [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "sha256-FSh3ww5KpktNSvqB3kcIaTxNHypwsVTHze0mgBtuJQE=";
+  };
+
+  cargoSha256 = "sha256-3DyLYegAzNbPpW6MUIDLm1QUqmGg8zH0Ps1dbdaSezs=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ Security ];
+
+  meta = with lib; {
+    description = "The fastest way to manage projects as tmux sessions";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/tmux-sessionizer/default.nix:27]: meta.homepage is missing

Comment (vinnymeller) [reply pkgs/tools/misc/tmux-sessionizer/default.nix:27]: added!

**Pull request merged**